command/CommandController.js

```diff
@@ -291,23 +317,56 @@ function getArticleMetadata(pocketArticle) {
   for (const auth in authors) {
     author = author ? `${author}, ${authors[auth].name}` : authors[auth].name;
   }
+  if (extendedData) {
+    let faviconPromise = faviconHelper.getWebsiteFavicon(
+      pocketArticle.resolved_url
+    );
 
-  return {
-    item_id: pocketArticle.item_id,
-    sort_id: pocketArticle.sort_id,
-    resolved_url: pocketArticle.resolved_url,
-    title: pocketArticle.resolved_title,
-    author,
-    lengthMinutes,
-    imageURL: pocketArticle.top_image_url
-  };
+    return faviconPromise
+      .then(function(faviconData) {
+        return {
+          item_id: pocketArticle.item_id,
+          sort_id: pocketArticle.sort_id,
+          resolved_url: pocketArticle.resolved_url,
+          title: pocketArticle.resolved_title,
+          author,
+          publisher: faviconData.website_name,
+          lengthMinutes,
+          imageURL: pocketArticle.top_image_url,
+          icon_url: faviconData.favicon_url
+        };
+      })
+      .catch(function() {
+        return {
+          item_id: pocketArticle.item_id,
+          sort_id: pocketArticle.sort_id,
+          resolved_url: pocketArticle.resolved_url,
+          title: pocketArticle.resolved_title,
+          author,
+          publisher: faviconHelper.getHostname(pocketArticle.resolved_url),
+          lengthMinutes,
+          imageURL: pocketArticle.top_image_url,
+          icon_url: ''
+        };
+      });
+  } else {
+    return {
+      item_id: pocketArticle.item_id,
+      sort_id: pocketArticle.sort_id,
+      resolved_url: pocketArticle.resolved_url,
+      title: pocketArticle.resolved_title,
+      author,
+      lengthMinutes,
+      imageURL: pocketArticle.top_image_url
+    };
```

Comment:
Definitely want to clean up these repeated object structures. I've also been favoring async/await over vanilla promises when we can. An approach like the following reads better, where you declare the object once and then add in the extended properties as appropriate:
```
  const result = {
    item_id: pocketArticle.item_id,
    sort_id: pocketArticle.sort_id,
    resolved_url: pocketArticle.resolved_url,
    title: pocketArticle.resolved_title,
    author,
    lengthMinutes,
    imageURL: pocketArticle.top_image_url
  };

  if (extendedData) {
    try {
      const faviconData = await faviconHelper.getWebsiteFavicon(
        pocketArticle.resolved_url
      );
      result.publisher = faviconData.website_name;
      result.icon_url = faviconData.favicon_url;
    } catch (err) {
      result.publisher = faviconHelper.getHostname(pocketArticle.resolved_url);
      result.icon_url = '';
    }
  }
  return result;
```

Reply:
Done